azure-pipelines.yml

```diff
@@ -764,3 +764,88 @@ stages:
           displayName: 'Install dependencies'
         - bash: python -m unittest discover -v test/ipynb
           displayName: 'Run image test'
+
+schedules:
+- cron: "0 6 * * *"
+  displayName: Nightly slow build
+  branches:
+    include:
+      - master
+  jobs:
+    - job: 'Linux_Slow_Tests'
+      pool: {vmImage: 'ubuntu-latest'}
+      strategy:
+        matrix:
+          Python37:
+            python.version: '3.8'
+      variables:
+        QISKIT_SUPPRESS_PACKAGING_WARNINGS: Y
+        PIP_CACHE_DIR: $(Pipeline.Workspace)/.pip
+        QISKIT_TEST_CAPTURE_STREAMS: 1
+        QISKIT_TESTS: run_slow
+      steps:
+        - task: UsePythonVersion@0
+          inputs:
+            versionSpec: '$(python.version)'
+          displayName: 'Use Python $(python.version)'
+        - task: Cache@2
+          inputs:
+            key: 'pip | "$(Agent.OS)" | "$(python.version)" |"$(Build.BuildNumber)"'
+            restoreKeys: |
+              pip | "$(Agent.OS)" | "$(python.version)"
+              pip | "$(Agent.OS)"
+              pip
+            path: $(PIP_CACHE_DIR)
+        - task: Cache@2
+          inputs:
+            key: 'stestr | "$(Agent.OS)" | "$(python.version)" | "$(Build.BuildNumber)"'
+            restoreKeys: |
+              stestr | "$(Agent.OS)" | "$(python.version)"
+              stestr | "$(Agent.OS)"
+              stestr
+            path: .stestr
+          displayName: Cache pip
+        - bash: |
+            set -e
+            python -m pip install --upgrade pip setuptools wheel virtualenv
+            virtualenv test-job
+            source test-job/bin/activate
+            pip install -U -r requirements.txt -r requirements-dev.txt -c constraints.txt
+            pip install -U -c constraints.txt .
+            pip install -U "qiskit-aer" "z3-solver" -c constraints.txt
+            python setup.py build_ext --inplace
+            sudo apt install -y graphviz
+            pip check
+          displayName: 'Install dependencies'
+        - bash: |
+            set -e
+            source test-job/bin/activate
+            export PYTHONHASHSEED=$(python -S -c "import random; print(random.randint(1, 4294967295))")
+            echo "PYTHONHASHSEED=$PYTHONHASHSEED"
+            stestr run
+          displayName: 'Run tests'
+        - task: CopyFiles@2
+          condition: failed()
+          displayName: 'Copy images'
+          inputs:
+            contents: '**/*.png'
+            targetFolder: $(Build.ArtifactStagingDirectory)
+        - task: PublishBuildArtifacts@1
+          condition: failed()
+          displayName: 'Publish images'
+          inputs:
+            pathtoPublish: '$(Build.ArtifactStagingDirectory)'
+            artifactName: 'drop_linux'
+        - bash: |
```

Comment:
Will this/can this generate any kind of push notification on failure?

Reply:
Not by default, azure's built in notifications are only for users with a linked azure devops account (which I think we both have), but they only offer a limit number of accounts to free organizations. So it's not a general solution (although you probably could personally subscribe to notifications for this https://docs.microsoft.com/en-us/azure/devops/notifications/about-notifications?view=azure-devops). I think the only way for this to work is to leverage the `tools/report_ci_failure.py` and run it in the job (and make sure we set the secret with the token for the @QacoBot account).